specs/phase0/beacon-chain.md

```diff
@@ -1550,36 +1546,33 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
         for operation in operations:
             fn(state, operation)
 
-    for_ops(body.proposer_slashings, process_proposer_slashing)
+    for_ops(body.equivocations, process_equivocation)
     for_ops(body.attester_slashings, process_attester_slashing)
     for_ops(body.attestations, process_attestation)
     for_ops(body.deposits, process_deposit)
     for_ops(body.voluntary_exits, process_voluntary_exit)
 ```
 
-##### Proposer slashings
-
-```python
-def process_proposer_slashing(state: BeaconState, proposer_slashing: ProposerSlashing) -> None:
-    header_1 = proposer_slashing.signed_header_1.message
-    header_2 = proposer_slashing.signed_header_2.message
-
-    # Verify header slots match
-    assert header_1.slot == header_2.slot
```

Comment:
Where does this check happen in `process_equivocation`?

Reply:
This check is done via the `equivocation_root `. Notice that the `slot` is part of the `equivocation_root`, and that both `signing_root_1` and `signing_root_2` use the same `equivocation_root`.